### PR TITLE
Allow user-specific terminal options

### PR DESCRIPTION
**Pull request by nevers**
Some boards, like the SparkFun ESP8266 Thing (https://www.sparkfun.com/products/13231) require pulling the DTR down in order to allow the device to flash or boot properly. 

This patch allows users to add extra parameters to miniterm, like for example --dtr=0

### CODE REVIEW

**Comment by slaff** [comment]
@nevers Please, add your change also to the other OS specific Makefile-xyz.mk files. Adding it only for Linux will leave us with inconsistent make system.

File: Sming/Makefile-linux.mk [hunk 1]
```diff
@@ -12,5 +12,5 @@ SDK_TOOLS	 ?= $(SDK_BASE)/tools
 ESPTOOL		 ?= $(ESP_HOME)/esptool/esptool.py
 KILL_TERM    ?= pkill -9 -f "$(COM_PORT) $(COM_SPEED_SERIAL)" || exit 0
 GET_FILESIZE ?= stat --printf="%s"
-TERMINAL     ?= python -m serial.tools.miniterm $(COM_PORT) $(COM_SPEED_SERIAL)
-MEMANALYZER  ?= $(OBJDUMP) -h -j .data -j .rodata -j .bss -j .text -j .irom0.text
\ No newline at end of file
+TERMINAL     ?= python -m serial.tools.miniterm $(COM_PORT) $(COM_SPEED_SERIAL) $(COM_OPTS)
+MEMANALYZER  ?= $(OBJDUMP) -h -j .data -j .rodata -j .bss -j .text -j .irom0.text
```
Comment (slaff) [inline Sming/Makefile-linux.mk]: Don't change file line/file ending.

File: Sming/Makefile-linux.mk [hunk 2]
```diff
@@ -12,5 +12,5 @@ SDK_TOOLS	 ?= $(SDK_BASE)/tools
 ESPTOOL		 ?= $(ESP_HOME)/esptool/esptool.py
 KILL_TERM    ?= pkill -9 -f "$(COM_PORT) $(COM_SPEED_SERIAL)" || exit 0
 GET_FILESIZE ?= stat --printf="%s"
-TERMINAL     ?= python -m serial.tools.miniterm $(COM_PORT) $(COM_SPEED_SERIAL)
-MEMANALYZER  ?= $(OBJDUMP) -h -j .data -j .rodata -j .bss -j .text -j .irom0.text
\ No newline at end of file
+TERMINAL     ?= python -m serial.tools.miniterm $(COM_PORT) $(COM_SPEED_SERIAL) $(COM_OPTS)
```
Comment (slaff) [inline Sming/Makefile-linux.mk]: Please, add your change also to the other OS specific Makefiles: Makefile-bsd.mk, Makefile-windows.mk, Makefile-macos.mk. Adding it only for Linux will leave us with inconsistent make system.

**Comment by nevers** [comment]
Right, forgot about the other build files. I'll have a look at that asap.

**Comment by nevers** [comment]
There you go.

**Comment by slaff** [comment]
@nevers You might have accidently committed also  wiki submodule changes. Please, remove that change from this PR.

**Comment by nevers** [comment]
I got to stop using the -a flag when committing. Fixed now, cheers.

**Comment by slaff** [comment]
@nevers Thanks!

**Pull request merged**